tests/ui/explicit-tail-calls/indexer.rs

```diff
@@ -0,0 +1,21 @@
+// Indexing taken from
+// https://github.com/phi-go/rfcs/blob/guaranteed-tco/text%2F0000-explicit-tail-calls.md#tail-call-elimination
+// should probably come back to after some decision on verbiage
```

Comment:
What do you mean by "decision on verbiage"?.. What is this test even testing?

you can make the test pass (hopefully) by just replacing `&dyn Fn(usize)` with `fn(usize)`